spring-kafka/src/main/java/org/springframework/kafka/support/DefaultKafkaHeaderMapper.java

```diff
@@ -321,7 +324,7 @@ else if (headerName.equals(KafkaHeaders.LISTENER_INFO) && matchesForInbound(head
 				headers.put(headerName, new String(header.value(), getCharset()));
 			}
 			else if (!(headerName.equals(JSON_TYPES)) && matchesForInbound(headerName)) {
-				if (jsonTypes != null && jsonTypes.containsKey(headerName)) {
+				if (jsonTypes.containsKey(headerName)) {
 					String requestedType = jsonTypes.get(headerName);
```

Comment:
we might well improve perf by using the following pattern:
```
String jsonType = jsonTypes.get(headerName);
if (jsonType != null) {
    ... ...
}
```
for we end up one map query, rather than twice (`containsKey()` and `get()`).